src/disp/program.c

```diff
@@ -819,6 +826,72 @@ static MVMuint32 value_index_how(MVMThreadContext *tc, MVMDispProgramRecording *
     return MVM_VECTOR_ELEMS(rec->values) - 1;
 }
 
+/* Ensures we have a values used entry for the specified HOW read. */
+static MVMuint32 value_index_unbox_int(MVMThreadContext *tc, MVMDispProgramRecording *rec,
+        MVMuint32 from_value) {
+    /* Look for an existing such value. */
+    MVMuint32 i;
+    for (i = 0; i < MVM_VECTOR_ELEMS(rec->values); i++) {
+        MVMDispProgramRecordingValue *v = &(rec->values[i]);
+        if (v->source == MVMDispProgramRecordingUnboxValue &&
+                v->how.from_value == from_value)
```

Comment:
I think the `->how` here is copy pasto, and it should be `->unbox`. Also I think it has to compare `kind` as well as `from_value`.